### PR TITLE
feat: support [Column]

### LINKED ISSUE #36: Support [Column]
Currently we support names (parameter and column) via `[DbValue]`, however there is also a well-known `[System.ComponentModel.DataAnnotations.Schema.ColumnAttribute]` that is commonly used.

I'm hesitant to use this *blindly*, because that would change existing behavior, so this should be triggered via a new `UseColumnAttributeAttribute` (yes, the naming is weird - see also `XmlAttributeAttribute`) which would apply at all levels (like most DapperAOT attributes):

- feature should only apply if `[UseColumnAttribute]` or `[UseColumnAttribute(true)]` is in play
- `[UseColumnAttribute(false)]` explicitly disables
- *only* `Name` would be used; we aren't interested in `Order` etc
- if we see a `[Column(...)]` with a `Name` but *don't* see any explicit on/off flag, emit a warning so the user knows to opt in/out
- works similar to `[DbValue]` with `Name`, but: doesn't impact parameters

See also https://stackoverflow.com/questions/77073286/can-dapper-mappings-be-automated-with-attributes

### PR DESCRIPTION
**Pull request by DeagleGross**
## Goal:
Support `[System.ComponentModel.DataAnnotations.Schema.ColumnAttribute]` as an additional way to override parameter name.
[Requested by some users (stackoverflow)](https://stackoverflow.com/questions/77073286/can-dapper-mappings-be-automated-with-attributes/77073456#77073456)

## Example:
To make override work, one need to specify both `[Column]` and `[UseColumnAttribute]` attributes, for example:
```csharp
class MyType
{
    [UseColumnAttribute]
    [Column("myOtherName")]
    public string Name { get; set; }
}
```

## Rules:
1. If `[Column]` and either `[UseColumnAttribute]` or `[UseColumnAttribute(true)]` is defined, than override will work
2. If `[UseColumnAttribute(false)]` is explicitly specified, then overriding will not happen
3. If `[Column]` is specified, but `[UseColumnAttribute]` is missing, then **DAP043** will be emitted to the user: "Attach the [UseColumnAttribute] attribute to make Dapper consider [Column]"
4. If _1)_ is followed, but also `[DbValue]` with property `Name` is specified, then a warning message will be emitted to the user: **DAP042** "A column name is specified via both [DbValue] and [Column]; '<dbvalue_name>' will be used"

Closes #36

### CODE REVIEW

File: src/Dapper.AOT.Analyzers/Internal/Inspection.cs [hunk 1]
```diff
@@ -800,7 +919,7 @@ private static bool TryGetAttributeValue<T>(AttributeData? attrib, string name,
                 int index = 0;
                 foreach (var p in ctor.Parameters)
                 {
-                    if (StringComparer.InvariantCultureIgnoreCase.Equals(p.Name == name))
```
Comment (mgravell) [inline src/Dapper.AOT.Analyzers/Internal/Inspection.cs:803]: oof, good catch; that must have been fun to find!

File: src/Dapper.AOT/UseColumnAttributeAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace Dapper
+{
+    /// <summary>
+    /// Specifies whether to use [System.ComponentModel.DataAnnotations.Schema.ColumnAttribute] for additional behavioral configuration
+    /// </summary>
+    public class UseColumnAttributeAttribute : Attribute
```
Comment (mgravell) [inline src/Dapper.AOT/UseColumnAttributeAttribute.cs:8]: what is the allowed usage here? can it be applied at the module level, or is it only respected when attached to the same type? are more than one allowed? can we add `[AttributeUsage(...)]` here accordingly

**Comment by mgravell** [comment]
looking good; fixed some minor nits:

- avoid `/en-us/` in docs links; if you cut it, it will use the user's locale settings
- formatting in docs
- used a different variant of that string equality fix

an open question on the new attribute, but: looking good!

File: src/Dapper.AOT.Analyzers/Internal/Inspection.cs (hunk 1 above)
Comment (DeagleGross) [reply src/Dapper.AOT.Analyzers/Internal/Inspection.cs:803]: you need to thank unit testing for such a treasure ;)

File: src/Dapper.AOT/UseColumnAttributeAttribute.cs (hunk 2 above)
Comment (DeagleGross) [reply src/Dapper.AOT/UseColumnAttributeAttribute.cs:9]: added such an expression.
`[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = false)]`

let me know if such `AttributeUsage` is correct

**Comment by DeagleGross** [comment]
> looking good; fixed some minor nits:
> 
> * avoid `/en-us/` in docs links; if you cut it, it will use the user's locale settings
> * formatting in docs
> * used a different variant of that string equality fix
> 
> an open question on the new attribute, but: looking good!

Thanks for the useful advices and fixed, Marc! Sorry for attribute usage - I added it. Ready for review

**Review by mgravell — APPROVED** [review]

**Pull request merged**